### PR TITLE
Add etcd encryption conf to profile IPI on AWS RHCOS_RHEL8.5_OVN_http_proxy

### PR DESCRIPTION
**Pull request by wangke19**
Add etcd encryption conf to profile IPI on AWS RHCOS_RHEL8.5_OVN_http_proxy.

### CODE REVIEW

**Comment by wangke19** [comment]
@jianlinliu My first prow PR for etcd encryption. Kindly review and give me feedback when you have free time. Thanks.

File: ci-operator/step-registry/etcd-encryption/conf/etcd-encryption-conf-ref.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    namespace: ocp
+    name: "4.12"
+    tag: upi-installer
+  commands: etcd-encryption-conf-commands.sh
+  resources:
+    requests:
+      cpu: 200m
+      memory: 100Mi
+  env:
+  - name: ZONES_COUNT
```
Comment (jianlinliu) [inline ci-operator/step-registry/etcd-encryption/conf/etcd-encryption-conf-ref.yaml:13]: Seem like the step does not need this env.

File: ci-operator/step-registry/etcd-encryption/conf/etcd-encryption-conf-ref.metadata.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+	"path": "etcd-encryption/conf/etcd-encryption-conf-ref.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/etcd-encryption/conf/etcd-encryption-conf-ref.metadata.json:2]: Go though etcd-encryption-conf-commands.sh, seem like no need `conf`subfolder, I suggest to remove the subfolder.

Comment (wangke19) [reply ci-operator/step-registry/etcd-encryption/conf/etcd-encryption-conf-ref.metadata.json]: Fixed

File: ci-operator/step-registry/etcd-encryption/conf/etcd-encryption-conf-ref.yaml (hunk 1 above)
Comment (wangke19) [reply ci-operator/step-registry/etcd-encryption/conf/etcd-encryption-conf-ref.yaml]: Removed env

**Comment by wangke19** [comment]
@jianlinliu Updated, please review when you have free time.

**Comment by jianlinliu** [comment]
Missing a validation job to verify this workflow.

**Comment by wangke19** [comment]
> Missing a validation job to verify this workflow.

@jianlinliu Could you give me one example? I can refer to that. Thanks.

**Comment by jianlinliu** [comment]
> @jianlinliu Could you give me one example? I can refer to that. Thanks.

example like https://github.com/openshift/release/pull/32093/files#diff-f4cc3edcda327e0da4b3bd773c06ee21831151125dc8d2e0a8e8dd32995f9159

**Comment by wangke19** [comment]
@jianlinliu I add the workflow to rehearse 4.12 config, and the installation and etcd encryption chain works as expect. Please help review, If OK. I will submit new commit to restore the changed  rehearse 4.12 config file.

**Comment by jianlinliu** [comment]
> I will submit new commit to restore the changed rehearse 4.12 config file

No need to restore, is is used for the testing purpose.

ci/prow/step-registry-metadata job failed, pls check its detail, follow it and fix it.

File: ci-operator/step-registry/etcd-encryption/etcd-encryption-commands.sh [hunk 3]
```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+
```
Comment (jianlinliu) [inline ci-operator/step-registry/etcd-encryption/etcd-encryption-commands.sh:2]: In prow ci, to be more restrict for shell syntax, generally we are setting, do you mind to follow the same rule.
```
set -o errexit
set -o nounset
set -o pipefail
```

**Comment by wangke19** [comment]
@jianlinliu Already updated, please help review when you have free time. Thanks.

**Pull request merged**